rca/static_src/javascript/programmes/programmes.api.js

```diff
@@ -0,0 +1,81 @@
+import PropTypes from 'prop-types';
+
+import { programmePage } from './programmes.types';
+
+const PROGRAMME_TYPE = 'programmes.ProgrammePage';
+const PROGRAMME_FIELDS = [
+    'degree_level',
+    'programme_description_subtitle',
+    'pathway_blocks',
+    'hero_image_square',
+];
+
+const WAGTAIL_API_ENDPOINT = '/api/v2/pages';
+
+/**
+ * Generates a Wagtail API query string based on the given attributes.
+ */
+export const getWagtailAPIQueryString = ({
+    search,
+    filters,
+    type,
+    fields,
+    limit,
+}) => {
+    const parameters = {
+        type: type || null,
+        limit: limit || null,
+        fields: fields.join(',') || null,
+        search: search || null,
+        ...filters,
+    };
+    Object.keys(parameters).forEach((param) => {
+        if (parameters[param] === null) {
+            delete parameters[param];
+        }
+    });
```

Comment:
This is probably a bit too smart just for the one API client, but I’m hoping we can reuse this on other projects in the future.